### PR TITLE
refactor: use better typing to remove any

### PR DESCRIPTION
**Pull request by HerringtonDarkholme**
## Summary

Remove unnecessary use of `any`

## Related issue (if exists)

None

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [x] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run changeset`.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: packages/rspack/src/rspack.ts [hunk 1]
```diff
@@ -124,7 +124,7 @@ function rspack(
 	options: MultiRspackOptions | RspackOptions,
 	callback?: Callback<Error, MultiStats> | Callback<Error, Stats>
 ) {
-	if (!asArray(options as any).every(i => rspackOptionsCheck(i))) {
+	if (!asArray(options).every(i => rspackOptionsCheck(i))) {
```
Comment (IWANABETHATGUY) [inline packages/rspack/src/rspack.ts:127]:
```suggestion
	if (!asArray(options).every(rspackOptionsCheck)) {
```
make the code more point free

**Review by IWANABETHATGUY — APPROVED** [review]

**Review by ahabhgk — APPROVED** [review]

File: packages/rspack/src/rspack.ts (hunk 1 above)
Comment (HerringtonDarkholme) [reply packages/rspack/src/rspack.ts:127]: Nope, rspackOptionsCheck curiously have overloaded signature that cannot use pointfree style.

**Comment by HerringtonDarkholme** [comment]
linux musl failed with core-dump. I personally don't think this is caused by this change. Any idea?

**Comment by IWANABETHATGUY** [comment]
> 

rerun should solve this issue,  we would dig into it and figure out why.

**Pull request merged**